File: backend/src/main/java/com/pgvaale/backend/entity/UserTiffin.java
package com.pgvaale.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

/** USER TIFFIN ENTITY */
@Entity
@Table(name = "user_tiffins")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserTiffin extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tiffin_id", nullable = false)
    private Tiffin tiffin;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RequestStatus status; // PENDING, ACCEPTED, REJECTED

    @Column(nullable = false)
    private LocalDateTime assignedDateTime; // When request was made

    private LocalDateTime deletionDateTime; // When service ends

    public enum RequestStatus {
        PENDING,
        ACCEPTED,
        REJECTED
    }
} 

File: PGVaaleDotNetBackend/Entities/UserTiffin.cs
using System.ComponentModel.DataAnnotations;

namespace PGVaaleDotNetBackend.Entities
{
    public class UserTiffin
    {
        public long Id { get; set; }

        [Required]
        public long UserId { get; set; }

        [Required]
        public long TiffinId { get; set; }

        [Required]
        public DateTime AssignedDateTime { get; set; }

        public DateTime? DeletionDateTime { get; set; }

        [Required]
        public RequestStatus Status { get; set; } = RequestStatus.PENDING;

        // Navigation properties
        public User User { get; set; } = null!;
        public Tiffin Tiffin { get; set; } = null!;

        public enum RequestStatus
        {
            PENDING,
            ACCEPTED,
            REJECTED
        }
    }
}